examples/devices/lighty-actions-device/src/main/java/io/lighty/netconf/device/action/processors/StartActionProcessor.java

```diff
@@ -41,34 +40,33 @@
 import org.slf4j.LoggerFactory;
 import org.w3c.dom.Element;
 
-public class StartActionProcessor extends ActionServiceDeviceProcessor {
+public final class StartActionProcessor extends ActionServiceDeviceProcessor {
 
     private static final Logger LOG = LoggerFactory.getLogger(StartActionProcessor.class);
 
-    private final CurrentAdapterSerializer adapterSerializer;
-    private ActionDefinition actionDefinition;
-    private Absolute actionPath;
     private final Start startAction;
+    private Absolute path;
+    private ActionDefinition definition;
+    private final CurrentAdapterSerializer adapterSerializer;
 
-    public StartActionProcessor(final Start startAction, final BindingDOMCodecServices codecServices) {
+    public StartActionProcessor(final Start startAction, final Absolute path, final ActionDefinition definition,
+            final BindingDOMCodecServices codecServices) {
         this.startAction = startAction;
-        final ConstantAdapterContext constantAdapterContext = new ConstantAdapterContext(codecServices);
-        this.adapterSerializer = constantAdapterContext.currentSerializer();
+        this.path = path;
+        this.definition = definition;
+        this.adapterSerializer = new ConstantAdapterContext(codecServices).currentSerializer();
     }
 
     @SuppressWarnings("checkstyle:IllegalCatch")
     @Override
     protected CompletableFuture<Response> execute(final Element requestXmlElement,
-            final Entry<Absolute, ActionDefinition> actionEntry) {
-        this.actionDefinition = actionEntry.getValue();
-        this.actionPath = actionEntry.getKey();
+            final ActionDefinition actionDefinition) {
```

Comment:
Please remove `final ActionDefinition actionDefinition`. It's redundant since it's already assigned in constructor.

Reply:
Done.